privacyidea/lib/caconnectors/localca.py

```diff
@@ -332,15 +332,14 @@ def _filename_from_x509(x509_name, file_extension="pem"):
         return a filename from the subject from an x509 object
         :param x509_name: The X509Name object
         :type x509_name: X509Name object
+        :param file_extension:
+        :type file_extension: str
         :return: filename
-        :rtype: basestring
+        :rtype: str
         """
         name_components = x509_name.get_components()
-        filename = ""
-        for (key, value) in name_components:
-            filename += value+"_"
-        filename = filename[:-1] + "." + file_extension
-        return filename
+        filename = "_".join([value.decode('utf8') for (key, value) in name_components])
```

Comment:
Here you are doing a ``decdode``. When should this be a ``to_unicode``?

Reply:
You are right, very inconsequential of me...

Reply:
fixed in 1fb2839